src/cloud/docker/docker-test-magecloud-pkg-code.md

```diff
@@ -0,0 +1,262 @@
+
+## Prepare the Docker environment for testing
+
+Before you run functional tests, you must set up and configure the test environment:
+
+1. Clone the GitHub repository for the package to test:
+
+   ```bash
+   git clone git@github.com:magento/<cloud-suite-package-repository>.git
+   ```
+
+   For example:
+
+   ```bash
+   git clone git@github.com:magento/ece-tools.git
+   ```
+
+1. Stop all services that use the following ports:
+
+   -  `80`—varnish
+   -  `443`—web, tls
+   -  `3306`—apache, mysql
+
+1. Add the following hostname to your `/etc/hosts` file.
+
+   ```conf
+   127.0.0.1 magento2.docker
+   ```
+
+   Alternatively, you can run the following command to add it to the file:
+
+   ```bash
+   echo "127.0.0.1 magento2.docker" | sudo tee -a /etc/hosts
+   ```
+
+1. Add required project dependencies if you are testing the following packages.
+
+   -  For the `{{site.data.var.mcp-package}}` package:
+
+      ```bash
+      composer require "magento/magento-cloud-docker:^1.0.0" --no-update
+      ```
+
+   -  For the `{{site.data.var.mcc-package}}` package:
+
+      ```bash
+      composer require "magento/magento-cloud-docker:^1.0.0" --no-update
+      composer require "magento/framework:*" --no-update
+      composer require "magento/module-store:*" --no-update
+      composer require "magento/module-url-rewrite:*" --no-update
+      ```
+
+1. Switch to the preferred PHP version for running tests.
+
+1. Update the project dependencies.
+
+   ```bash
+   composer update
+   ```
+
+### Add credentials
+
+Add credentials to your Docker environment using any of the following methods:
+
+-  Use environment variables
+-  Load credentials from the environment configuration file
+-  Add variables directly to the test configuration file
+
+{:.procedure}
+To add credentials using environment variables:
+
+1. Create the following environment variables:
+
+   ```bash
+   export REPO_USERNAME=your_public_key
+   ```
+
+   ```bash
+   export REPO_PASSWORD=your_private_key
+   ```
+
+1. To install some packages from GitHub, create an environment variable for the GitHub token:
+
+   ```bash
+   `export GITHUB_TOKEN=your_github_token
+   ````
+
+{:.procedure}
+To load credentials from the environment configuration file:
+
+1. Run the following commands to write credentials to the ./.env file.
+
+   ```bash
+   echo "REPO_USERNAME=your_public_key" >> ./.env
+   ```
+
+   ```bash
+   echo "REPO_PASSWORD=your_private_key" >> ./.env
+   ```
+
+1. Edit the `codeception.dist.yml` file:
+
+   ```yaml
+   params:
+       - tests/functional/configuration.dist.yml
+       - env
+       - .env
+   ```
+
+1. Add a dependency for the [vlucas/phpdotenv] package required to load the environment variables.
+
+   ```bash
+   composer require "vlucas/phpdotenv": "^3.0"
+   ```
+
+{:.procedure}
+To add credentials directly to the test configuration file:
+
+1. Edit the `codeception.dist.yml` file.
```

Comment:
I made it a two step procedure to accomodate the  {:.procedure} class title so that this section would be formatted the same as the others.  Initially, these three procedures were all combined into a single procedure.